### ISSUE TITLE
Add support for postgresql Source

### ISSUE THREAD

**Issue by tobalsgithub** [RuntimeTools/appmetrics#60]
My team uses postgresql as our database, but we'd love to give this package a try. Any chance you'd accept a PR for postgresql support?

**Comment by seabaylea** [comment]
@tobalsgithub as I happens, I'm aware that there's a prototype for postgres support that's being worked on in Dan Cunnington's fork:
https://github.com/DanCunnington/appmetrics

@DanCunnington it might be worth you syncing up with TC to see if your prototype covers what his team is looking for.

**Comment by DanCunnington** [comment]
Hi @tobalsgithub, currently this postgres probe monitors the 'query' method of the pg module (https://www.npmjs.com/package/pg) and returns a JSON response:

**Comment by DanCunnington** [comment]
Hi @tobalsgithub, currently this probe monitors the 'query' method of the pg module (https://www.npmjs.com/package/pg). 

For each query made, a 'postgres' event is emitted detailing the start time of the query, the SQL query itself and the duration of the query (ms). 

Here is a test case based on the query example in the module docs (https://www.npmjs.com/package/pg) with the added appmetrics code: 

```js
var appmetrics = require('appmetrics');

//Live monitoring
var amapi = appmetrics.monitor();

//Log all postgres events
amapi.on('postgres', function(response) {
  var readableDate = new Date(response.time);
  console.log("Time of query: "+readableDate.toString());
  console.log("SQL Query: "+response.query);
  console.log("Duration: "+response.duration);
});

//Example from module docs
var pg = require('pg');
var conString = "postgres://user:password@host/database";
 
var client = new pg.Client(conString);

client.connect(function(err) {
  if(err) {
    return console.error('could not connect to postgres', err);
  }
  client.query('SELECT NOW() AS "theTime"', function(err, result) {
    if(err) {
      return console.error('error running query', err);
    }

    console.log(result.rows[0].theTime);
    //End
    client.end();
  });
});
```

The 'postgres' event emitted will contain the details of the query, similar to the following:

```js
 { time: 1447336180602,
  query: SELECT NOW() AS "theTime",
  duration: 2.854412 }
```

Feel free to add any suggestions you may have.

Thanks

**Comment by tobalsgithub** [comment]
Hey @DanCunnington, this looks like exactly what we need. Very similar to how the mysql probe works, which is what I would expect. Do you have plans to create a PR any time soon?

**Comment by DanCunnington** [comment]
It's pretty much ready to go now @tobalsgithub, I'm going to conduct further testing and try to gain some more feedback if possible. It should be ready within a week or so but feel free to pull down my code for now if it meets your requirements.

**Comment by seabaylea** [comment]
In that case its a likely candidate for the upcoming 1.0.4 release.

**Comment by tobalsgithub** [comment]
Sounds awesome. Appreciate the work!

**Comment by seabaylea** [comment]
This should be resolved by PR #66 

@tobalsgithub if you'd like to test, you can how install from the master branch.

**Comment by seabaylea** [comment]
Closing based on PR #66. We can re-open if there's issues.
